### PR TITLE
Add qwen3vl

### PR DESCRIPTION
**Pull request by Matvezy**
# Description

Add Qwen3vl for inference and also use platform trained model

## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Tested in staging

## Any specific deployment considerations

No

## Docs

-   [ ] Docs updated? What were the changes: No

### CODE REVIEW

File: requirements/requirements.transformers.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # TODO: update to 2.8.0 once pre-built flashattn is available
 torch>=2.0.1,<2.7.0
 torchvision>=0.15.0
-transformers>=4.53.3,<4.57.0
+transformers>=4.57.3,<v5.0.0rc0
```
Comment (grzegorz-roboflow) [inline requirements/requirements.transformers.txt]: will this result in allowing `transformers==v5.0.0`?

File: inference_experimental/tests/e2e_platform_tests/test_qwen3vl_e2e.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    model = AutoModel.from_pretrained("qwen3vl-2b-instruct")
+
+    # WHEN
+    captions = model.prompt(images=dog_image_numpy, prompt="What is in the image?")
+
+    # THEN
+    assert isinstance(captions, list)
+    assert len(captions) == 1
+    assert isinstance(captions[0], str)
+    assert len(captions[0]) > 0
```
Comment (grzegorz-roboflow) [inline inference_experimental/tests/e2e_platform_tests/test_qwen3vl_e2e.py:20]: Is result produced by the model unstable? Why not asserting the actual string produced by the model?

File: inference/models/README.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ The models supported by Roboflow Inference have their own licenses. View the lic
 | `inference/models/grounding_dino`     | [Apache 2.0](https://github.com/IDEA-Research/GroundingDINO/blob/main/LICENSE)       |              👍              |
 | `inference/models/owlv2`              | [Apache 2.0](https://github.com/google-research/scenic/blob/main/LICENSE)            |              👍              |
 | `inference/models/paligemma`          | [Apache 2.0](https://github.com/google-research/big_vision/blob/main/LICENSE), [Gemma](https://ai.google.dev/gemma/terms) |              👍              |
-| `inference/models/qwen25vl(7b)`       | [Apache 2.0](https://github.com/QwenLM/Qwen2.5-VL/blob/main/LICENSE)                 |              👍              |
+| `inference/models/qwen25vl(7b)`       | [Apache 2.0](https://github.com/QwenLM/Qwen3-VL/blob/main/LICENSE)                 |              👍              |
+| `inference/models/qwen3vl(2b)`       | [Apache 2.0](https://github.com/QwenLM/Qwen2.5-VL/blob/main/LICENSE)                 |              👍              |
```
Comment (grzegorz-roboflow) [inline inference/models/README.md]: looks like rows got mixed, `qwen25` is pointing to `qwen3` license, and `qwen3` is pointing to `qwen25` license

File: inference/core/env.py [hunk 4]
```diff
@@ -191,6 +191,8 @@
 
 QWEN_2_5_ENABLED = str2bool(os.getenv("QWEN_2_5_ENABLED", True))
 
+QWEN_3_ENABLED = str2bool(os.getenv("QWEN_3_ENABLED", True))
```
Comment (grzegorz-roboflow) [inline inference/core/env.py:194]: QWEN_2_5_ENABLED is manually disabled in jetson Dockerfile, should this be disabled too?

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+##########################################################################
+# Qwen3-VL Workflow Block Manifest
+##########################################################################
+class BlockManifest(WorkflowBlockManifest):
+    # Qwen3-VL only needs an image and an optional text prompt.
+    images: Selector(kind=[IMAGE_KIND]) = ImageInputField
+    prompt: Optional[str] = Field(
+        default=None,
+        description="Optional text prompt to provide additional context to Qwen3-VL. Otherwise it will just be None",
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py]: Instead of `Otherwise it will just be None` would be nicer to mention what will be the model outcome in case user does not provide prompt

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        examples=["What is in this image?"],
+    )
+
+    # Standard model configuration for UI, schema, etc.
+    model_config = ConfigDict(
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:35]: minor - most blocks define `model_config` and `type` on top of class properties, lets stick to that convention

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        json_schema_extra={
+            "name": "Qwen3-VL",
+            "version": "v1",
+            "short_description": "Run Qwen3-VL on an image.",
+            "long_description": (
+                "This workflow block runs Qwen3-VL—a vision language model that accepts an image "
+                "and an optional text prompt—and returns a text answer based on a conversation template."
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:42]: What will happen if prompt is not provided?

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+        system_prompt: Optional[str],
+    ) -> BlockResult:
+        # Convert each image to the format required by the model.
+        inference_images = [
+            i.to_inference_format(numpy_preferred=False) for i in images
+        ]
+        # Use the provided prompt (or an empty string if None) for every image.
+        prompt = prompt or ""
+        system_prompt = system_prompt or ""
+        prompts = [prompt + "<system_prompt>" + system_prompt] * len(inference_images)
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:164]: do we also need `"</system_prompt>"`?

File: tests/inference/integration_tests/qwen_test.py [hunk 9]
```diff
@@ -33,4 +33,32 @@ def test_qwen25_inference(server_url: str, clean_loaded_models_fixture) -> None:
...
+)
+def test_qwen3_inference(server_url: str, clean_loaded_models_fixture) -> None:
+    # given
+    payload = {
+        "api_key": doc_api_key,
+        "image": {
+            "type": "url",
+            "value": "https://media.roboflow.com/dog.jpeg",
+        },
+        "prompt": "Tell me something about this dog!<system_prompt>You are a helpful assistant.",
```
Comment (grzegorz-roboflow) [inline tests/inference/integration_tests/qwen_test.py:51]: would be nice to also have a test case with empty prompt

File: inference_experimental/inference_exp/models/qwen3vl/qwen3vl_hf.py [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+)
+from peft import PeftModel
+from transformers import (
+    AutoProcessor,
+    BitsAndBytesConfig,
+    Qwen3VLForConditionalGeneration,
+)
+
+
+class Qwen3VLHF:
```
Comment (grzegorz-roboflow) [inline inference_experimental/inference_exp/models/qwen3vl/qwen3vl_hf.py:25]: Implementation in `inference` specfies `cache` property, is cache not needed here?

File: requirements/requirements.transformers.txt (hunk 1 above)
Comment (Matvezy) [reply requirements/requirements.transformers.txt]: Good point, I think you're right! Changing

File: inference_experimental/tests/e2e_platform_tests/test_qwen3vl_e2e.py (hunk 2 above)
Comment (Matvezy) [reply inference_experimental/tests/e2e_platform_tests/test_qwen3vl_e2e.py:20]: Yes can assert the string, should be deterministic

File: inference/models/README.md (hunk 3 above)
Comment (Matvezy) [reply inference/models/README.md]: Fixed!

File: inference/core/env.py (hunk 4 above)
Comment (Matvezy) [reply inference/core/env.py:194]: Yes!

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py (hunk 5 above)
Comment (Matvezy) [reply inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py]: Cool, change this

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py (hunk 6 above)
Comment (Matvezy) [reply inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:35]: Fixed

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py (hunk 7 above)
Comment (Matvezy) [reply inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:42]: We just send empty string, fixed it to have sensible defaults.

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py (hunk 8 above)
Comment (Matvezy) [reply inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:164]: Yes it's a special token needed to point out system prompt for qwen.

File: inference_experimental/inference_exp/models/qwen3vl/qwen3vl_hf.py (hunk 10 above)
Comment (Matvezy) [reply inference_experimental/inference_exp/models/qwen3vl/qwen3vl_hf.py:25]: My understanding is inference-experimental models don't need it? Since they don't manage their own artifact downloading/caching.

File: inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py (hunk 8 above)
Comment (Matvezy) [reply inference/core/workflows/core_steps/models/foundation/qwen3vl/v1.py:164]: *Special string

**Comment by grzegorz-roboflow** [comment]
testing Dockerfile builds:

 - [ ] https://github.com/roboflow/inference/actions/runs/20725012128
 - [ ] https://github.com/roboflow/inference/actions/runs/20725029501
 - [ ] https://github.com/roboflow/inference/actions/runs/20725034519

**Review by grzegorz-roboflow — APPROVED** [review]
